zaza/openstack/utilities/openstack.py

```diff
@@ -1769,6 +1793,63 @@ def create_image(glance, image_url, image_name, image_cache_dir=None, tags=[]):
     return image
 
 
+def create_volume(cinder, size, name=None, image=None):
+    """Create cinder volume.
+    :param cinder: Authenticated cinderclient
+    :type cinder: cinder.Client
+    :param size: Size of the volume
+    :type size: int
+    :param name: display name for new volume
+    :type name: Option[str, None]
+    :param image: Image to download to volume.
+    :type image: Option[str, None]
+    :returns: cinder volume pointer
+    :rtype: cinderclient.common.utils.RequestIdProxy
+    """
+    logging.debug('Creating volume')
+    if not size:
+        raise Exception("Size for volume not specified")
```

Comment:
In general you would not use ``Exception`` class directly, create a new exception class so we know where the failure occurred.

However, in this case it might be sufficient to just pass the value or non-value of ``size`` on to cinder and let it raise an exception, or does it fail silently?

Reply:
I'm removing the Exception as it will be raised anyway.
Tried calling with no argument:
> TypeError: create_volume() missing 1 required positional argument: 'size'

Calling with "" or None as size:
> cinderclient.exceptions.BadRequest: Invalid input received: Volume size '' must be an integer and greater than 0 (HTTP 400)